index.js

```diff
@@ -4,26 +4,40 @@ var vt = require('vector-tile');
 var request = require('request');
 var Protobuf = require('pbf');
 var format = require('util').format;
+var fs = require('fs');
 
 module.exports = function(args, callback) {
-    var zxy = args.url.match(/\/(\d+)\/(\d+)\/(\d+)/);
 
-    args.x = args.x === undefined ? zxy[2] : args.x;
-    args.y = args.y === undefined ? zxy[3] : args.y;
-    args.z = args.z === undefined ? zxy[1] : args.z;
-
-    if (args.x === undefined || args.y === undefined || args.z === undefined) {
-        return callback(new Error(format("Could not determine tile z, x, and y from %s; specify manually with -z <z> -x <x> -y <y>", JSON.stringify(url))));
+    try {
+        var file = fs.lstatSync(args.uri);
+        if (file.isFile()) {
+            fs.readFile(args.uri, function(err, data) {
+                if (err) throw err;
+                readTile(data);
+            });
+        }
+    } catch(err) {
+        request.get({
+            url: args.uri,
+            gzip: true,
+            encoding: null
+        }, function (err, response, body) {
+            if (err) throw err;
+            readTile(body);
+        });
     }
```

Comment:
1. No, just do `fs.exits` before `fs.readFile`
2. Right, you need to use `parsed.pathname`. I would about to comment that you don't need to use `host` until I realized - likely what you did - that passing in a relative path to `file://` like `file://./hello.txt` leads to `host:'.'` and `pathname:'/hello.txt'` so combining those does look needed! Btw, they look like they will work the same no matter if `./hello.txt` is passed or `file://./hello.txt` so I think you can drop the `if (parsed.protocol && parsed.protocol.indexOf('file') > -1) {` logic and just do `fs.readFile(parsed.host + parsed.pathname, function(err, data) {`